ci/test/04_install.sh

```diff
@@ -58,6 +58,7 @@ else
     bash -c "export PATH=$BASE_SCRATCH_DIR/bins/:\$PATH && cd $P_CI_DIR && $*"
   }
 fi
+export -f DOCKER_EXEC
```

Comment:
This part I don't understand. It was working "correctly" before (without retry).

Reply:
Besides [Bash Reference Manual](https://www.gnu.org/software/bash/manual/bash.html), there are more references:
- https://unix.stackexchange.com/questions/22796/can-i-export-functions-in-bash
- https://stackoverflow.com/questions/1885871/exporting-a-function-in-shell

If shell function `retry` is used, it runs a subshell, and cannot see definitions of other shell functions unless they are exported: "Functions may be exported so that subshells automatically have them defined with the `-f` option to the `export` builtin." (from  [Bash Reference Manual](https://www.gnu.org/software/bash/manual/bash.html)).

This could be verified by `set -x`.